### DISCUSSION TITLE
Is it valid to only calculate SHAP for a subset of features? i.e. exclude some features and treat them as context

### DISCUSSION THREAD

**Discussion by lysiabean** [shap/shap#2269] [Q&A]
I have a ML model that I'm currently using KernelExplainer to calculate SHAP values. However, for this 1 feature, I want to hold it constant and don't want to attribute any credits to it. I want to treat it as part of the given context.

Has anyone done this before and is it valid from the theoretical perspective?

Thanks in advance!

**Comment by jainraj** [comment]
Really good question. See the [Explaining a Series of Models by Propagating Shapley Values - Chen et al. - Section 3.2](https://arxiv.org/pdf/2105.00108.pdf) and [The Explanation Game: Explaining Machine Learning Models Using Shapley Values - Merrick and Taly](https://arxiv.org/pdf/1909.08128.pdf) papers. Both do a good job of analysing what it could mean to change the background datasets - basically the kind of question we ask from the model changes and thus the SHAP values need to be understood in that context.
